src/query/string_opfunc.c

```diff
@@ -3113,6 +3120,12 @@ db_json_array (DB_VALUE * result, DB_VALUE * arg[], int const num_args)
 
   for (int i = 0; i < num_args; i++)
     {
+      if (DB_IS_NULL (arg[i]))
+	{
+	  db_json_add_element_to_array (new_doc, (JSON_DOC *) NULL);
```

Comment:
Is cast to JSON_DOC * necessary?

Reply:
yes, otherwise, it gets confused whether to call add_element (char *) or add_element (JSON_DOC *)